File: go/internal/models/userModel.go
package models

type User struct {
	//gorm.Model
	ID           uint           `gorm:"primary_key; not null" json:"id"`
	FirstName    string         `gorm:"size:64; not null; unique" json:"first_name"`
	LastName     string         `gorm:"size:64; not null" json:"last_name"`
	Photo        string         `gorm:"size:255;" json:"photo"`
	Username     string         `gorm:"size:64; not null; unique " json:"username"`
	Password     string         `gorm:"size:255; not null" json:"password"`
	Email        string         `gorm:"size:100; not null; unique" json:"email"`
	Address      string         `gorm:"size:255; not null" json:"address"`
	RoleID       uint           `gorm:"size: 64; default:2; not null" json:"role_id"`
	GroupeVoyage []GroupeVoyage `gorm:"many2many:groupe_members;"json:"groupe_voyage"`
	Voyage       []Voyage       `gorm:"foreignKey:UserId;" json:"voyage"`
}

File: flutter_app/lib/models/user_model.dart
class User {
  int id;
  String firstName;
  String lastName;
  String username;
  String? photo;
  String email;
  String? password;
  String address;
  int roleId;

  User({
    required this.id,
    this.photo,
    required this.firstName,
    required this.lastName,
    required this.username,
    required this.email,
    this.password,
    required this.address,
    required this.roleId,
  });

  @override
  String toString() {
    return 'User{id: $id, first_name: $firstName, last_name: $lastName, '
        'username: $username, email: $email, password: $password, photo: $photo,'
        'address: $address, role_id: $roleId}';
  }

  User.fromJson(Map<String, dynamic> json)
      : id = json['id'] ?? 0,
        firstName = json['first_name'] ?? '',
        lastName = json['last_name'] ?? '',
        username = json['username'] ?? '',
        password = json['password'] ?? '',
        photo = json['photo'] ?? '',
        email = json['email'] ?? '',
        address = json['address'] ?? '',
        roleId = json['role_id'] ?? 0;

  Map<String, dynamic> toJson() {
    Map<String, dynamic> value = {
      'id': id,
      'first_name': firstName,
      'last_name': lastName,
      'username': username,
      'password': password,
      'photo': photo,
      'email': email,
      'address': address,
      'role_id': roleId
    };
    value['id'] = id;
    return value;
  }
}
